oioswift/common/middleware/container_hierarchy.py

```diff
@@ -603,6 +607,47 @@ def update_copy_headers(self, req, env2):
             req.headers['Oio-Copy-From'] = '/' + c_container + '/' + c_obj
             env2['HTTP_OIO_COPY_FROM'] = '/' + c_container + '/' + c_obj
 
+    def _handle_dir_object(self, req, start_response, account, container,
+                           path, missing=False):
+        if req.method in ('GET', 'HEAD'):
+            is_available = False
+            if not missing:
+                key = self.key(account, container, OBJ, path)
+                if self.redis_keys_format == REDIS_KEYS_FORMAT_V1:
+                    is_available = self.conn.exists(key + "/")
+                else:
+                    is_available = self.conn.hexists(key, path + "/")
+
+            if is_available:
+                # add a fake AllUsers permissions, otherwise it will
+                # be dropped by ACL checks in S3 Layer
+                oheaders = {
+                    'Content-Length': 0,
+                    'Etag': 'd41d8cd98f00b204e9800998ecf8427e',
+                    'x-object-sysmeta-swift3-acl': FAKE_ACL}
+                start_response("200 OK", oheaders.items())
+            else:
+                start_response("404 Not Found", {}.items())
+            return ['']
+
+        if req.method == 'PUT':
+            # TODO (MBO) we should accept create key d1/d2/ only
+            # for empty object
+            key = self.key(account, container, OBJ, path)
+            self._create_placeholder(req, account, container,
+                                     OBJ, path + '/')
+            oheaders = {'Content-Length': 0,
+                        'Etag': 'd41d8cd98f00b204e9800998ecf8427e'}
```

Comment:
To avoid repeating this magic string, we can use `swift.common.utils.MD5_OF_EMPTY_STRING`.

Reply:
:+1: